### PR TITLE
feat(registry): add openscad-language-server

### LINKED ISSUE #797: [New package]: openscad-LSP
### I've searched open & closed issues for similar requests

Yes

### Package name

openscad-LSP

### Package homepage

https://github.com/Leathong/openscad-LSP

### Languages

openscad

### How is this package distributed?

https://github.com/Leathong/openscad-LSP

### PR DESCRIPTION
**Pull request by calexanderberg**
Hi I decided to add the openscad lsp written by Leathong in Rust. It is an attempt to close #797 
If I find the time I'll also pull a request for nvim-lspconfig.

Link to lsp: https://github.com/Leathong/openscad-LSP

If there are any issues with this I would love to improve it.

### CODE REVIEW

**Comment by calexanderberg** [comment]
#797

File: lua/mason-registry/openscad-language-server/init.lua [hunk 1]
```diff
@@ -0,0 +1,13 @@
+local Pkg = require "mason-core.package"
+local cargo = require "mason-core.managers.cargo"
+
+return Pkg.new {
+    name = "openscad-language-server",
+    desc = [[Language Server Protocol implementation for OpenSCAD, written in Rust.]],
+    homepage = "https://github.com/Leathong/openscad-LSP",
+    languages = { Pkg.Lang.Openscad },
```
Comment (williamboman) [inline lua/mason-registry/openscad-language-server/init.lua:8]:
```suggestion
    languages = { Pkg.Lang.OpenSCAD },
```

**Comment by williamboman** [comment]
@williambotman /fixup

**Review by williamboman — APPROVED** [review]
Cool!

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/lua/mason-registry/index.lua b/lua/mason-registry/index.lua
index 2a02498..59e23b4 100644
--- a/lua/mason-registry/index.lua
+++ b/lua/mason-registry/index.lua
@@ -155,7 +155,7 @@ return {
   omnisharp = "mason-registry.omnisharp",
   ["omnisharp-mono"] = "mason-registry.omnisharp-mono",
   ["opencl-language-server"] = "mason-registry.opencl-language-server",
-  ["openscad-language-server"] = "mason-registry.openscad-language-server",
+  ["openscad-lsp"] = "mason-registry.openscad-lsp",
   perlnavigator = "mason-registry.perlnavigator",
   ["php-cs-fixer"] = "mason-registry.php-cs-fixer",
   ["php-debug-adapter"] = "mason-registry.php-debug-adapter",
diff --git a/lua/mason-registry/openscad-language-server/init.lua b/lua/mason-registry/openscad-lsp/init.lua
similarity index 91%
rename from lua/mason-registry/openscad-language-server/init.lua
rename to lua/mason-registry/openscad-lsp/init.lua
index 814c4c0..1307127 100644
--- a/lua/mason-registry/openscad-language-server/init.lua
+++ b/lua/mason-registry/openscad-lsp/init.lua
@@ -2,7 +2,7 @@ local Pkg = require "mason-core.package"
 local cargo = require "mason-core.managers.cargo"
 
 return Pkg.new {
-    name = "openscad-language-server",
+    name = "openscad-lsp",
     desc = [[Language Server Protocol implementation for OpenSCAD, written in Rust.]],
     homepage = "https://github.com/Leathong/openscad-LSP",
     languages = { Pkg.Lang.Openscad },
diff --git a/lua/mason/mappings/language.lua b/lua/mason/mappings/language.lua
index 61bab7a..a836b4e 100644
--- a/lua/mason/mappings/language.lua
+++ b/lua/mason/mappings/language.lua
@@ -88,7 +88,7 @@ return {
   onescript = { "bsl-language-server" },
   opencl = { "opencl-language-server" },
   openfoam = { "foam-language-server" },
-  openscad = { "openscad-language-server" },
+  openscad = { "openscad-lsp" },
   perl = { "perlnavigator" },
   php = { "intelephense", "php-cs-fixer", "php-debug-adapter", "phpactor", "phpcbf", "phpcs", "phpmd", "phpstan", "pint", "psalm", "semgrep" },
   powershell = { "powershell-editor-services" },
```

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/lua/mason-registry/index.lua b/lua/mason-registry/index.lua
index 2a02498..59e23b4 100644
--- a/lua/mason-registry/index.lua
+++ b/lua/mason-registry/index.lua
@@ -155,7 +155,7 @@ return {
   omnisharp = "mason-registry.omnisharp",
   ["omnisharp-mono"] = "mason-registry.omnisharp-mono",
   ["opencl-language-server"] = "mason-registry.opencl-language-server",
-  ["openscad-language-server"] = "mason-registry.openscad-language-server",
+  ["openscad-lsp"] = "mason-registry.openscad-lsp",
   perlnavigator = "mason-registry.perlnavigator",
   ["php-cs-fixer"] = "mason-registry.php-cs-fixer",
   ["php-debug-adapter"] = "mason-registry.php-debug-adapter",
diff --git a/lua/mason-registry/openscad-language-server/init.lua b/lua/mason-registry/openscad-lsp/init.lua
similarity index 91%
rename from lua/mason-registry/openscad-language-server/init.lua
rename to lua/mason-registry/openscad-lsp/init.lua
index ae988b9..1be3803 100644
--- a/lua/mason-registry/openscad-language-server/init.lua
+++ b/lua/mason-registry/openscad-lsp/init.lua
@@ -2,7 +2,7 @@ local Pkg = require "mason-core.package"
 local cargo = require "mason-core.managers.cargo"
 
 return Pkg.new {
-    name = "openscad-language-server",
+    name = "openscad-lsp",
     desc = [[Language Server Protocol implementation for OpenSCAD, written in Rust.]],
     homepage = "https://github.com/Leathong/openscad-LSP",
     languages = { Pkg.Lang.OpenSCAD },
diff --git a/lua/mason/mappings/language.lua b/lua/mason/mappings/language.lua
index 61bab7a..a836b4e 100644
--- a/lua/mason/mappings/language.lua
+++ b/lua/mason/mappings/language.lua
@@ -88,7 +88,7 @@ return {
   onescript = { "bsl-language-server" },
   opencl = { "opencl-language-server" },
   openfoam = { "foam-language-server" },
-  openscad = { "openscad-language-server" },
+  openscad = { "openscad-lsp" },
   perl = { "perlnavigator" },
   php = { "intelephense", "php-cs-fixer", "php-debug-adapter", "phpactor", "phpcbf", "phpcs", "phpmd", "phpstan", "pint", "psalm", "semgrep" },
   powershell = { "powershell-editor-services" },
```

**Comment by williamboman** [comment]
Renamed the package `openscad-language-server -> openscad-lsp` to match upstream name

**Comment by williamboman** [comment]
Ported to https://github.com/mason-org/mason-registry/pull/282 which will soon:tm: be the new registry source.

**Pull request merged**